google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/impl/PaginatedRequest.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.bigquery.impl
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.dispatch.ExecutionContexts
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.Uri.Query
+import akka.http.scaladsl.unmarshalling.{FromEntityUnmarshaller, Unmarshal}
+import akka.http.scaladsl.util.FastFuture
+import akka.stream.alpakka.googlecloud.bigquery.impl.http.BigQueryHttp
+import akka.stream.alpakka.googlecloud.bigquery.scaladsl.Paginated
+import akka.stream.alpakka.googlecloud.bigquery.{BigQueryAttributes, BigQueryException, BigQuerySettings}
+import akka.stream.scaladsl.Source
+import akka.{Done, NotUsed}
+
+import scala.concurrent.Future
+
+@InternalApi
+private[bigquery] object PaginatedRequest {
+
+  def apply[Out: FromEntityUnmarshaller](request: HttpRequest, initialPageToken: Option[String])(
+      implicit paginated: Paginated[Out]
+  ): Source[Out, NotUsed] =
+    Source
+      .fromMaterializer { (mat, attr) =>
+        implicit val system = mat.system
+        implicit val settings = BigQueryAttributes.resolveSettings(attr, mat)
+
+        Source.unfoldAsync[Either[Done, Option[String]], Out](Right(initialPageToken)) {
+          case Left(Done) =>
+            FastFuture.successful(None)
```

Comment:
There's probably a reason for `FastFuture` in akka-http, but it seems we could just use  `Future.successful(None)` here. We could avoid multiple allocations by defining it at the object scope and just returning it when the pages are done.